java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -431,15 +429,14 @@ public void register(Class<?> cls, int classId) {
     Preconditions.checkArgument(classId >= 0 && classId < Short.MAX_VALUE);
     Preconditions.checkArgument(
         !extRegistry.registeredClassIdMap.containsKey(cls),
```

Comment:
We can use `if` instead of `Preconditions#checkArgument`, thereby reducing the number of `extRegistry.registeredClassIdMap.get`.

Reply:
Could you elabrate more? I don't see any difference here

Reply:
```java
    Preconditions.checkArgument(
        !extRegistry.registeredClassIdMap.containsKey(cls),
        "Class %s already registered with id %s.",
        cls,
        extRegistry.registeredClassIdMap.get(cls));
    Preconditions.checkArgument(
        !extRegistry.registeredClasses.containsKey(cls.getName()),
        "Class %s with name %s has been registered, registering class with same name are not allowed.",
        extRegistry.registeredClasses.get(cls.getName()),
        cls.getName());
```

```java
    if (extRegistry.registeredClassIdMap.containsKey(cls)) {
      throw new IllegalArgumentException(
          String.format(
              "Class %s already registered with id %s.",
              cls, extRegistry.registeredClassIdMap.get(cls)));
    }

    if (extRegistry.registeredClasses.containsKey(cls.getName())) {
      throw new IllegalArgumentException(
          String.format(
              "Class %s with name %s has been registered, registering class with same name are not allowed.",
              extRegistry.registeredClasses.get(cls.getName()), cls.getName()));
    }
```

The former `extRegistry.registeredClassIdMap.get` will definitely be executed, while the latter will only be executed when there is an exception. We can save the cost of `extRegistry.registeredClassIdMap.get`.

Reply:
Good suggestion, updated